### PR TITLE
Make Varnish triggers invalidate by Surrogate-Key - fixes #6773

### LINKED ISSUE #6773: Make invalidation triggers calculate and invalidate using Surrogate-Key instead of Surrogate-Key

### PR DESCRIPTION
**Pull request by zenitraM**
:bomb:

Blocked by Windshaft-cartodb and SQL API releases

### CODE REVIEW

**Comment by zenitraM** [comment]
cc @rochoa, @rambo, should we make this configurable so it is able to install the old trigger compatible with old tiler versions?

also cc @CartoDB/systems as [this](https://github.com/CartoDB/cartodb/pull/6805/files#diff-8312ad0561ef661716b48d09478362f3R3) will need to be done on onprems at some point during the upgrade process

File: app/models/user/db_service.rb [hunk 1]
```diff
@@ -1143,6 +1143,7 @@ def create_function_invalidate_varnish_telnet
                 trigger_verbose = #{varnish_trigger_verbose}
 
                 client = GD.get('varnish', None)
+                import hashlib, base64
```
Comment (rochoa) [inline app/models/user/db_service.rb]: Should we cache this modules as we do with varnish client?

Comment (rochoa) [inline app/models/user/db_service.rb]: I see we don't do this in the other method.

Comment (luisbosque) [inline app/models/user/db_service.rb]: +1

Comment (zenitraM) [inline app/models/user/db_service.rb]: It seems imports are kept between python blocks executed on the same session but not kept as variables:

```
postgres=# DO $$ import sys; plpy.warning(sys.modules.keys()) $$ language plpythonu;
WARNING:  ['plpy', 'copy_reg', 'sre_compile', '_hashlib', '_sre', 'encodings', 'site', '__builtin__', 'sysconfig', '__main__', 'encodings.encodings', 'abc', 'posixpath', 'spiexceptions', '_weakrefset', 'errno', 'encodings.codecs', 'sre_constants', 're', '_abcoll', 'types', '_codecs', '_warnings', 'genericpath', 'stat', 'zipimport', 'encodings.__builtin__', 'warnings', 'UserDict', 'encodings.utf_8', 'sys', 'pwd', 'codecs', 'os.path', 'hashlib', 'sitecustomize', 'signal', 'traceback', 'apport_python_hook', 'linecache', 'posix', 'encodings.aliases', 'exceptions', 'sre_parse', 'os', '_weakref']
CONTEXT:  PL/Python anonymous code block
DO
postgres=# DO $$ import base64 $$ language plpythonu;
DO
postgres=# DO $$ import sys; plpy.warning(sys.modules.keys()) $$ language plpythonu;
WARNING:  ['plpy', 'copy_reg', 'sre_compile', '_hashlib', '_sre', 'encodings', 'site', '__builtin__', 'sysconfig', '__main__', 'encodings.encodings', 'abc', 'posixpath', 'spiexceptions', '_weakrefset', 'errno', 'base64', 'binascii', 'encodings.codecs', 'sre_constants', 're', '_abcoll', 'types', '_codecs', '_struct', '_warnings', 'genericpath', 'stat', 'zipimport', 'encodings.__builtin__', 'warnings', 'UserDict', 'encodings.utf_8', 'sys', 'pwd', 'codecs', 'os.path', 'struct', 'hashlib', 'sitecustomize', 'signal', 'traceback', 'apport_python_hook', 'linecache', 'posix', 'encodings.aliases', 'exceptions', 'sre_parse', 'os', '_weakref']
CONTEXT:  PL/Python anonymous code block
DO

postgres=# DO $$ print base64 $$ language plpythonu;
ERROR:  NameError: global name 'base64' is not defined
CONTEXT:  Traceback (most recent call last):
  PL/Python anonymous code block, line 1, in <module>
    print base64
PL/Python anonymous code block
```

So I'll add some caching

**Comment by zenitraM** [comment]
Now caching all imports on both Varnish methods

File: app/models/user/db_service.rb [hunk 2]
```diff
@@ -1202,21 +1200,16 @@ def create_function_invalidate_varnish_http
...
-                    #
-                    # TODO: do not invalidate responses with surrogate key
-                    #       "not_this_one" when table "this" changes :/
-                    #       --strk-20131203;
-                    #
-                    client = httplib.HTTPConnection('#{varnish_host}', #{varnish_port}, False, timeout)
-                    client.request('PURGE', '/batch', '', {"Invalidation-Match": ('^#{@user.database_name}:(.*%s.*)|(cdb_tablemetadata)|(table)$' % table_name.replace('"',''))  })
+                    client = GD['httplib'].HTTPConnection('#{varnish_host}', #{varnish_port}, False, timeout)
+                    cache_key = "t:" + GD['base64'].b64encode(GD['hashlib'].sha256('#{@user.database_name}:%s' % table_name).digest())[0:6]
+                    client.request('PURGE', '/key', '', {"Invalidation-Match": ('\\\\b%s\\\\b' % cache_key) })
```
Comment (ethervoid) [inline app/models/user/db_service.rb]: Why don't we use here `purge_command = Cartodb::config[:varnish_management]["purge_command"]`

**Pull request merged**